src/loaders/jobs/taskfarmer/taskfarmer_task_mgr.py

```diff
@@ -0,0 +1,164 @@
+import os
+import shutil
+from enum import Enum
+
+import pandas as pd
+
+import src.loaders.jobs.taskfarmer.taskfarmer_common as tf_common
+
+NERSC_SLURM_DONE_FILE = 'done.tasks.txt.tfin'
+
+
+class JobStatus(Enum):
+    """
+    Job status.
+    """
+    PENDING = 'PENDING'
+    RUNNING = 'RUNNING'
+    COMPLETED = 'COMPLETED'
+    FAILED = 'FAILED'
+
+
+class TFTaskManager:
+
+    def __init__(self, kbase_collection, load_ver, tool, root_dir):
+        self.kbase_collection = kbase_collection
+        self.load_ver = load_ver
+        self.tool = tool
+        self.root_dir = root_dir
+
+        self.job_dir = self.get_job_dir()
+        self.task_exists = os.path.isdir(self.job_dir)
+        self.tasks_df = self.retrieve_task_info()
+
+    def get_job_dir(self):
+        """
+        Get the job directory for the kbase_collection, load_ver, and tool.
+
+        The job directory is named as <kbase_collection>_<load_ver>_<tool>.
+        """
+        job_dir = os.path.join(self.root_dir, tf_common.TASKFARMER_JOB_DIR,
+                               f'{self.kbase_collection}_{self.load_ver}_{self.tool}')
+
+        return job_dir
+
+    def recreate_job_dir(self):
+        """
+        Recreate the job directory.
+        """
+
+        print(f'recreating job dir {self.job_dir}')
+        if os.path.exists(self.job_dir):
+            shutil.rmtree(self.job_dir)
+        os.makedirs(self.job_dir, exist_ok=True)
+
+        return self.job_dir
+
+    def create_job_dir(self):
+        """
+        Create the job directory.
+        """
+        os.makedirs(self.job_dir, exist_ok=True)
+
+        return self.job_dir
+
+    def get_task_info_file(self):
+        """
+        Get the task info file.
+
+        The task info file is located under the TASKFARMER_JOB_DIR directory.
+        The file name is retrieved from TASK_INFO_FILE.
+        """
+        task_info_file = os.path.join(self.root_dir, tf_common.TASKFARMER_JOB_DIR, tf_common.TASK_INFO_FILE)
+
+        if not os.path.exists(task_info_file):
+            with open(task_info_file, 'w') as f:
+                pass  # creates an empty file
+
+        return task_info_file
```

Comment:
What actually writes to the task info file? I would suggest rather than having some other code write to it directly, have a method in this class that the code calls so that the file name / location / format can be changed in the future in this class without breaking outside code

Reply:
👍

Reply:
:+1: